java/src/main/java/genericpubsub/PublishStream.java

```diff
@@ -52,24 +52,30 @@ public PublishStream(ExampleConfigurations exampleConfigurations) {
      * @return ByteString
      * @throws Exception
      */
-    public void publishStream(int numEventsToPublish) throws Exception {
+    public void publishStream(int numEventsToPublish, Boolean singleBatch) throws Exception {
         CountDownLatch finishLatch = new CountDownLatch(1);
         AtomicReference<CountDownLatch> finishLatchRef = new AtomicReference<>(finishLatch);
         final List<Status> errorStatuses = Lists.newArrayList();
-        final List<PublishResponse> publishResponses = Lists.newArrayListWithExpectedSize(numEventsToPublish);
+        final int numExpectedPublishResponses = singleBatch ? 1 : numEventsToPublish;
+        final List<PublishResponse> publishResponses = Lists.newArrayListWithExpectedSize(numExpectedPublishResponses);
         AtomicInteger failed = new AtomicInteger(0);
         StreamObserver<PublishResponse> pubObserver = getDefaultPublishStreamObserver(errorStatuses, finishLatchRef,
-                numEventsToPublish, publishResponses, failed);
+                numExpectedPublishResponses, publishResponses, failed);
 
         // construct the stream
         requestObserver = (ClientCallStreamObserver<PublishRequest>) asyncStub.publishStream(pubObserver);
 
-        for (int i = 0; i < numEventsToPublish; i++) {
-            requestObserver.onNext(generatePublishRequest(i));
+        if (singleBatch == false) {
+            // Publish each event in a separate batch
+            for (int i = 0; i < numEventsToPublish; i++) {
+                requestObserver.onNext(generatePublishRequest("0055f000005mc66AAA", i, singleBatch));
```

Comment:
Do you think it would be possible to store the user ID in the arguments.yaml file and pass it in as an argument, which then gets passed in to generateProducerEvent/generateProducerEvents and createEventMessage/createEventMessages?